### PR TITLE
[docs] [train] Example Gallery Train examples to be the same as the ones on the Train Examples page

### PR DESCRIPTION
**Pull request by angelinalg**
The Example Gallery with the training tag selected generates more examples than is in the Train Examples page. Removing some mislabelled examples by removing the `training` tag. Adding in two missing examples:

- templates/04_finetuning_llms_with_deepspeed/ (Fine-tuning Llama-2 models)
- /train/examples/transformers/transformers_torch_trainer_basic

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-overview/examples.rst [hunk 1]
```diff
@@ -1173,7 +1186,7 @@ Ray Examples
         Simple Parallel Model Selection
 
     .. grid-item-card:: :bdg-secondary:`Code example`
-        :class-item: gallery-item core training
```
Comment (pcmoritz) [inline doc/source/ray-overview/examples.rst:1176]: I think we should remove this whole `ray-core/examples/plot_example-lm.html` example, the docs views are very low (also nowadays the `transformers` library is a better way to train these models than `fair-seq` :)

File: doc/source/ray-overview/examples.rst [hunk 2]
```diff
@@ -1062,7 +1075,7 @@ Ray Examples
         The Magic of Merlin - Shopify's New ML Platform
 
     .. grid-item-card:: :bdg-success:`Tutorial`
-        :class-item: gallery-item training
```
Comment (pcmoritz) [inline doc/source/ray-overview/examples.rst:1065]: `Tutorial` is a misnomer for this one -- it is a talk (albeit an interesting one)

Any proposal how to handle this @matthewdeng ?

File: doc/source/ray-overview/examples.rst [hunk 3]
```diff
@@ -104,21 +104,21 @@ Ray Examples
         Aviary toolkit serving live traffic for LLMs
 
     .. grid-item-card:: :bdg-success:`Tutorial`
-        :class-item: gallery-item pytorch training
```
Comment (pcmoritz) [inline doc/source/ray-overview/examples.rst:107]: Intuitively, the `train` tag on this is right, @matthewdeng is there a reason why we don't include it in the Train examples page?

File: doc/source/ray-overview/examples.rst [hunk 4]
```diff
@@ -104,21 +104,21 @@ Ray Examples
...
+        :class-item: gallery-item pytorch
         :link: /train/examples/pytorch/convert_existing_pytorch_code_to_ray_train
         :link-type: doc
 
         Get started with Ray Train from an existing PyTorch codebase
 
     .. grid-item-card:: :bdg-secondary:`Code example`
-        :class-item: gallery-item training
```
Comment (pcmoritz) [inline doc/source/ray-overview/examples.rst:114]: Intuitively, the `training` tag on this is right, @matthewdeng is there a reason why we don't include it in the Train examples page?

File: doc/source/ray-overview/examples.rst [hunk 5]
```diff
@@ -104,21 +104,21 @@ Ray Examples
...
+        :class-item: gallery-item
         :link: /train/examples/lightgbm/lightgbm_example
         :link-type: doc
 
         Distributed training with LightGBM
 
     .. grid-item-card:: :bdg-success:`Tutorial`
-        :class-item: gallery-item training
```
Comment (pcmoritz) [inline doc/source/ray-overview/examples.rst:121]: Intuitively, the `training` tag on this is right, @matthewdeng is there a reason why we don't include it in the Train examples page?

**Comment by pcmoritz** [comment]
Thanks for doing this, this looks great (I made some comments).

One thing that is a little unresolved: On the many model training, ideally we would have one way we recommend that works well and remove the others. Since this is a training workload, maybe @matthewdeng should make the call here.

View numbers from the docs (Sept 1 2023 - Dec 3 2023):

- tune/examples/batch_tuning.html (our current recommended way per https://docs.ray.io/en/master/ray-overview/use-cases.html#how-do-i-do-many-model-training-on-ray): 381
- data/examples/batch_training: 1425
- /ray-core/examples/batch_training: 703

File: doc/source/ray-overview/examples.rst (hunk 2 above)
Comment (matthewdeng) [reply doc/source/ray-overview/examples.rst:1065]: Maybe we could add `Talk` as an example type? cc @simran-2797

File: doc/source/ray-overview/examples.rst (hunk 3 above)
Comment (matthewdeng) [reply doc/source/ray-overview/examples.rst:107]: @woshiyyya do you remember the reason for not including this?

File: doc/source/ray-overview/examples.rst (hunk 5 above)
Comment (matthewdeng) [reply doc/source/ray-overview/examples.rst:121]: For Ray 2.7 we made a decision that Ray Train examples would only contain the Torch ecosystem examples.

**Comment by matthewdeng** [comment]
For many model training, my personal preference would be to recommend Ray Core. It works well and offers the most about of flexibility for the user, which I think is practical for many model training because it doesn't have a one-size-fits-all solution.

File: doc/source/ray-overview/examples.rst (hunk 3 above)
Comment (woshiyyya) [reply doc/source/ray-overview/examples.rst:107]: Overall, I think it makes sense to add the training label here. This is not included in the Ray Train-Examples page, probably because we already have the "Getting Started with PyTorch" user guide and want users to go there.

File: doc/source/ray-overview/examples.rst (hunk 1 above)
Comment (angelinalg) [reply doc/source/ray-overview/examples.rst:1176]: Ack. Removing.

File: doc/source/ray-overview/examples.rst (hunk 3 above)
Comment (angelinalg) [reply doc/source/ray-overview/examples.rst]: Ok - I'll leave it in.

File: doc/source/ray-overview/examples.rst (hunk 5 above)
Comment (angelinalg) [reply doc/source/ray-overview/examples.rst:121]: Ok - I can keep this in the Example Gallery for now. @pcmoritz suggested that as a follow up we can remove additional examples when we feel like the Guide is complete enough to stand alone. Make sense?

**Comment by angelinalg** [comment]
Based on @matthewdeng's position about Many Model Training, shall I remove the `/tune/examples/batch_tuning` card that is called, "Batch Training and Tuning using Ray Tune`?

**Comment by angelinalg** [comment]
I will address the Many Model Training issue in another PR.

File: doc/source/tune/examples/ml-frameworks.rst [hunk 6]
```diff
@@ -103,10 +103,3 @@ At the end of these guides you will often find links to even more examples.
 
             End-to-end Example for Tuning a TensorFlow Model
 
-    .. grid-item-card::
```
Comment (woshiyyya) [inline doc/source/tune/examples/ml-frameworks.rst:106]: From our offline discussion, we should keep this example but just remove the "training" tag in the example library? (but it seems that there is no `training` tag originally)

So I think we should keep it in the Tune example page.

**Review by woshiyyya — CHANGES_REQUESTED** [review]

File: doc/source/ray-overview/examples.rst [hunk 7]
```diff
@@ -1131,21 +1123,7 @@ Ray Examples
...
-    .. grid-item-card:: :bdg-success:`Tutorial`
-        :class-item: gallery-item pytorch computer-vision data inference
-        :link: /data/examples/batch_inference_object_detection
-        :link-type: doc
-
-        Object Detection Batch Inference with PyTorch FasterRCNN_ResNet50
-
-    .. grid-item-card:: :bdg-success:`Tutorial`
-        :class-item: gallery-item data data-processing training
+        :class-item: gallery-item data data-processing
```
[the remark this answers is not in the document]
Comment (angelinalg) [reply doc/source/ray-overview/examples.rst:1126]: Ah, good question. I also wondered this and will follow up with a resolution. In short, `data` is for the library. I will remove the `data-processing` tag in a separate PR.

File: doc/source/tune/examples/ml-frameworks.rst (hunk 6 above)
Comment (angelinalg) [reply doc/source/tune/examples/ml-frameworks.rst:106]: Ah yes. Good catch. I forgot to put this back in. Will you follow up with moving the example to the `tune/examples` directory, @woshiyyya ?

Comment (angelinalg) [reply doc/source/tune/examples/ml-frameworks.rst]: Just pushed a commit to add this back in.

Comment (woshiyyya) [reply doc/source/tune/examples/ml-frameworks.rst]: Ok I'll post a PR to move it to `tune/examples`.

File: doc/source/ray-overview/examples.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (woshiyyya) [reply doc/source/ray-overview/examples.rst:1126]: OK Thank you!

**Review by woshiyyya — APPROVED** [review]

**Review by pcmoritz — APPROVED** [review]

**Review by angelinalg — COMMENTED** [review]
Removing comment.
